pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,433 @@
+const TOKEN_MIN_BALANCE: Balance = 10_000;
+
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+#[drink::test(sandbox = Sandbox)]
+fn new_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	// Token exists after the deployment.
+	assert!(session.sandbox().asset_exists(&TOKEN_ID));
+	let expected = Created {
+		id: TOKEN_ID,
+		creator: account_id_from_slice(contract.as_ref()),
+		admin: account_id_from_slice(contract.as_ref()),
+	}
+	.encode();
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice()); // Successfully emit event.
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn existing_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+
+	// Fails to deploy contract with a non-existing token ID.
+	assert!(
+		deploy_with_existing_constructor(&mut session, BundleProvider::local()?, TOKEN_ID).is_err()
+	);
+	expect_deployment_reverted(&session, PSP22Error::Custom(String::from("Unknown")));
+
+	// Successfully deploy contract with an existing token ID.
+	let actor = session.get_actor();
+	session.sandbox().create(&TOKEN_ID, &actor, TOKEN_MIN_BALANCE).unwrap();
+	deploy_with_existing_constructor(&mut session, BundleProvider::local()?, TOKEN_ID)?;
+
+	assert!(session.sandbox().asset_exists(&TOKEN_ID));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+	assert_eq!(session.sandbox().total_supply(&TOKEN_ID), 0);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn balance_of_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	const AMOUNT: Balance = TOKEN_MIN_BALANCE * 4;
+	// Mint tokens.
+	assert_ok!(session.sandbox().mint_into(&TOKEN_ID, &ALICE, AMOUNT));
+
+	// Successfully return a correct balance.
+	assert_eq!(balance_of(&mut session, ALICE), AMOUNT);
+	assert_eq!(balance_of(&mut session, ALICE), session.sandbox().balance_of(&TOKEN_ID, &ALICE));
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn mint_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	const AMOUNT: Balance = TOKEN_MIN_BALANCE * 4;
+
+	// Successfully mint tokens.
+	assert_ok!(mint(&mut session, ALICE, 0));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+	assert_eq!(session.sandbox().total_supply(&TOKEN_ID), 0);
+	assert_eq!(session.sandbox().balance_of(&TOKEN_ID, &ALICE), 0);
```

Comment:
Move to the bottom and remove total supply test. This is such a small piece and is heavily tested. Not important to the example

Reply:
comment above should be like burn `No-op`. I also liked that you only used 2 lines there.